tests/plugins/autocomplete/viewposition.js

```diff
@@ -152,6 +152,26 @@
 
 			assert.areEqual( '200px', view.element.getStyle( 'top' ), 'View is displayed below the caret' );
 			assert.areEqual( '50px', view.element.getStyle( 'left' ) );
+		},
+
+		// (#3582)
+		'test view position beyond window borders': function( editor ) {
+			var orginalWindowHeight = editor.element.getWindow().getViewPaneSize().height;
+
+			// Simulate small window height.
+			editor.element.getWindow().$.innerHeight = 200;
+
+			var view = createPositionedView( editor, {
+				caretRect: { top: 100, bottom: 110, left: 50 },
```

Comment:
You are missing `right` here (which you added in this PR too). This causes some `NaN`s in the positioning code:

![image](https://user-images.githubusercontent.com/1061942/95205616-e670f600-07e5-11eb-85ee-286e5882899d.png)